website/docs/registry/run-tasks.mdx

```diff
@@ -1,15 +1,16 @@
 ---
 page_title: Run Tasks
 description: |-
- Add your run task to the public Terraform Registry and browse available run tasks and their documentation.
+  Add your run task to the public Terraform Registry and browse available run tasks and their documentation.
 ---
 
 # Run Tasks
-Terraform Cloud [run tasks](/terraform/cloud-docs/workspaces/settings/run-tasks) let you directly integrate third-party tools and services into Terraform runs. 
 
-Each run task in a [workspace](/terraform/cloud-docs/workspaces) sends information to an external service at a specific run stage, such as after Terraform completes the plan. The external service processes the information and then sends a passed or failed status back to Terraform Cloud. Depending on the workspace’s configuration, failed run tasks can stop runs and prevent Terraform from provisioning infrastructure. 
+Terraform Cloud [run tasks](/terraform/cloud-docs/workspaces/settings/run-tasks) let you directly integrate third-party tools and services at certain stages in the Terraform Cloud run lifecycle.
 
-For example, the [HCP Packer validation run task](/hcp/docs/packer/manage-image-use/terraform-cloud-run-tasks) checks the image artifacts within a Terraform configuration. The run task fails if the configuration references images marked as unusable.
+Each run task in a [workspace](/terraform/cloud-docs/workspaces) sends information to an external service at a specific run stage. The external service processes the information and returns a passed or failed status to Terraform Cloud. For example, you could add a run task that analyzes a plan's output before Terraform applies that plan.
+
+Depending on the workspace’s configuration, failed run tasks can stop runs and prevent Terraform from provisioning infrastructure. For example, the [HCP Packer validation run task](/hcp/docs/packer/manage-image-use/terraform-cloud-run-tasks) checks the image artifacts within a Terraform configuration. The run task fails if the configuration references images marked as unusable.
```

Comment:
```suggestion
Each run task in a [workspace](/terraform/cloud-docs/workspaces) sends run data to an external service at a specific run stage. The external service processes the data and returns a passed or failed status to Terraform Cloud. For example, you could add a run task that checks configuration for any security vulnerabilities before Terraform applies that plan.

Depending on the enforcement level, failed run tasks can stop runs and prevent Terraform from provisioning infrastructure. For example, the [HCP Packer validation run task](/hcp/docs/packer/manage-image-use/terraform-cloud-run-tasks) checks the image artifacts within a Terraform configuration. The run task fails if the configuration references images marked as unusable.
```

the last two sentences of the second paragraph about the hcp packer run task feel more specific to a run task itself rather than the enforcement level. maybe a rephrase there, or just use that as the example for the first paragraph instead?

Reply:
Taking your suggestion and adding a second specific example of mandatory vs. advisory tasks to better explain the concept of enforcement level: 
```suggestion
Each run task in a [workspace](/terraform/cloud-docs/workspaces) sends run data to an external service at a specific run stage. The external service processes the data and returns a passed or failed status to Terraform Cloud. For example, you could add a run task that checks configuration for any security vulnerabilities before Terraform applies that plan.

Depending on the enforcement level, failed run tasks can stop runs and prevent Terraform from provisioning infrastructure. For example, a run fails if a mandatory task fails, even if an advisory task succeeds. If an advisory task fails, but a mandatory task succeeds, the run succeeds and proceeds to the apply stage, where it can provision infrastructure.
```